site/app/controllers/grading/ElectronicGraderController.php

```diff
@@ -947,10 +953,21 @@ public function saveSingleComponent() {
             $user_ids_to_grade = array_map(function(User $user) { return $user->getId(); }, $users_to_grade);
             if (!in_array($user_id, $user_ids_to_grade)) {
                 $this->core->addErrorMessage("You do not have permission to grade {$user_id}");
-                return;
+
+                $response = array('status' => 'failure');
+                $this->core->getOutput()->renderJson($response);
+                return $response;
             }
         }
+        /*
+        if ($gradeable->getCurrentVersion() === null) {
+            $this->core->addErrorMessage("Cannot grade an assignment with no submission.");
```

Comment:
Why did you comment this out?

Reply:
This warning would appear on every page, and because the grading rubric now says no submission and does not allow for grading, I didn't think it was needed